yarn-project/p2p/src/services/peer-manager/peer_manager.ts

```diff
@@ -268,6 +318,10 @@ export class PeerManager implements PeerManagerInterface {
   }
 
   public getPeerScore(peerId: string): number {
+    if (this.config.p2pAllowOnlyValidators && !this.isAuthenticatedPeer(peerIdFromString(peerId))) {
```

Comment:
If we require auth and a peer is not auth'd, it is given a `-Infinity` peer score. This ensure it won't be used by gossipsub.

Reply:
Are there any implications of returning -Inf while the authentication handshake is happening?